Makefile

```diff
@@ -864,6 +864,16 @@ ifneq ("$(TOUCHID_TAG)", "")
 		| gotestsum --raw-command -- cat
 endif
 
+# Runs benchmarks once to make sure they pass.
+# This is intended to run in CI during unit testing to make sure benchmarks don't break.
+# To limit noise and improve speed this will only run on packages that have benchmarks.
+# Race detection is not enabled because it significantly slows down benchmarks.
+# todo: Use gotestsum when it is compatible with benchmark output. Currently will consider all benchmarks failed.
+.PHONY: test-go-bench
+test-go-bench: PACKAGES = $(shell grep --exclude-dir api --include "*_test.go" -lr testing.B .  | xargs dirname | xargs go list)
```

Comment:
you could put a `sort -u` in the pipeline that would remove the duplicate directories (which is not really necessary as go test will not test the same package even if specified multiple times) but also make the order deterministic. Also, I think the `go list` is not necessary as you can run the tests by package path and does not need the fully-qualified package name.